Car.cpp

```diff
@@ -0,0 +1,64 @@
+#include "Car.hpp"
+#include <iostream>
+
+Car::Car() {
+    std::cout << __FUNCTION__ << " - ";
+}
+
+Car::~Car() {
+    std::cout << __FUNCTION__ << std::endl;
+}
+
+void Car::turnLeft() {
+    std::cout << __FUNCTION__ << std::endl;
+}
+
+void Car::turnRight() {
+    std::cout << __FUNCTION__ << std::endl;
+}
+
+void Car::brake(int speed) {
+    std::cout << __FUNCTION__ << ": ";
+    try {
+        if (speed > currentSpeed_) {
+            throw BreakeToHigherSpeedError(std::to_string(currentSpeed_));
+        }
+        setSpeed(speed);
+    } catch (BreakeToHigherSpeedError& speed) {
+        std::cerr << speed.what() << "\n";
+    }
+}
+
+void Car::accelerate(int speed) {
+    std::cout << __FUNCTION__ << ": ";
+    try {
+        if (speed < currentSpeed_ && speed > -20) {
+            throw AccelerateToLoweSpeedError(std::to_string(currentSpeed_));
+        }
+        setSpeed(speed);
+    } catch (AccelerateToLoweSpeedError& speed) {
+        std::cerr << speed.what() << "\n";
+    }
+}
+
+void Car::setSpeed(int speed) {
+    try {
+        if (speed < -20) {
+            throw ToLowSpeedError(std::to_string(speed));
+        }
+        if (speed > 200) {
```

Comment:
Good direction,but why -20 and 200? That values should be car values.

Reply:
I also dont understand these values. Below zero speed means you are going backward using reverse gear? I suggest you to avoid magic numbers. Make them constexpr or const